spring-pulsar/src/main/java/org/springframework/pulsar/listener/DefaultPulsarMessageListenerContainer.java

```diff
@@ -240,15 +253,172 @@ else if (messageListener != null) {
 						.maxNumMessages(containerProperties.getMaxNumMessages())
 						.maxNumBytes(containerProperties.getMaxNumBytes())
 						.timeout(containerProperties.getBatchTimeoutMillis(), TimeUnit.MILLISECONDS).build();
+
+				Set<String> topicNames = (Set<String>) propertiesToConsumer.remove("topicNames");
```

Comment:
If we end up going w/ this approach I think we should add a comment about why we are pulling these 2 fields out of the map (it is not too hard to figure this out below as they are 1st class method params but it could be helpful to have a simple comment as such).

Reply:
done